lib/DxilContainer/DxcContainerBuilder.cpp

```diff
@@ -161,6 +164,13 @@ DxcContainerBuilder::SerializeContainer(IDxcOperationResult **ppResult) {
       errorHeap.Detach();
     }
 
+    // Add Hash.
+    LPVOID PTR = pResult->GetBufferPointer();
+    if (!IsDxilContainerLike(PTR, pResult->GetBufferSize()))
+      return E_FAIL;
+
+    HashAndUpdate((DxilContainerHeader *)PTR);
```

Comment:
Actually, IsDxilContainerLike should never fail here, and it returns the `DxilContainerHeader *` upon success.  Plus, we need to make sure validation succeeded before hashing:
```suggestion
    if (SUCCEEDED(valHR))
      HashAndUpdate(IsDxilContainerLike(pResult->GetBufferPointer(),
                                        pResult->GetBufferSize()));
```